swap/src/bitcoin/wallet.rs

```diff
@@ -146,12 +145,13 @@ impl Wallet {
             .or_insert_with(|| {
                 let (sender, receiver) = watch::channel(ScriptStatus::Unseen);
                 let client = self.client.clone();
+                let sync_interval = self.sync_interval;
 
                 tokio::spawn(async move {
                     let mut last_status = None;
 
                     loop {
-                        tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
NACK this "simplification".

This patch removes an essential aspect of the bandwidth optimization because now, every individual subscription will result in a network call whereas before, we only refreshed the local state based on the ping interval, regardless of how often the internal subscription queried the local state.

We need the "barrier" inside `update_state` to ensure we don't scale the number of network requests with the number of subscriptions.

Reply:
Thanks! I knew that there was a reason that the ping was inside the client but when I looked at it yesterday I just did not see the benefit anymore. Did not think about multiple subscriptions ;)

I find it a bit hard to understand, because the subscription triggers the update (which is effectively only triggered if the current update interval expired). I think it is an efficient design, but it is not that easy to grasp why it is designed this way. I thought about handling the updates in a separate thread, but I am not sure that this complete decoupling would make it easier to understand.

Reply:
One thing that could make things easier to understand is moving all the code that is spawned for each Subscription to a dedicated fn on subscription. That might make it more obvious that it happens per subscription whereas the client exists only once as can be seen by the Arc<Mutex>.
Not sure what you are getting at with threads, we have tasks available and already use them to run these things concurrently. `tokio::spawn` is effectively the `thread::spawn` you are thinking of I believe.